PyInstaller/hooks/rthooks/pyi_rth_pkgres.py

```diff
@@ -9,14 +9,179 @@
 # SPDX-License-Identifier: Apache-2.0
 #-----------------------------------------------------------------------------
 
+import os
+import sys
+import pathlib
 
-import pkg_resources as res
+import pkg_resources
 from pyimod03_importers import FrozenImporter
 
+SYS_PREFIX = pathlib.PurePath(sys._MEIPASS)
 
-# To make pkg_resources work with froze moduels we need to set the 'Provider'
+
+# To make pkg_resources work with frozen modules we need to set the 'Provider'
 # class for FrozenImporter. This class decides where to look for resources
 # and other stuff. 'pkg_resources.NullProvider' is dedicated to PEP302
 # import hooks like FrozenImporter is. It uses method __loader__.get_data() in
 # methods pkg_resources.resource_string() and pkg_resources.resource_stream()
-res.register_loader_type(FrozenImporter, res.NullProvider)
+
+# We subclass the NullProvider and implement _has(), _isdir(), and _listdir(),
+# which are needed for pkg_resources.resource_exists(), resource_isdir(),
+# and resource_listdir() to work. We cannot use the DefaultProvider,
+# because it provides filesystem-only implementations (and overrides _get()
+# with a filesystem-only one), whereas our provider needs to also support
+# embedded resources.
+
+class _TocFilesystem:
+    def __init__(self, toc_files, toc_dirs=[]):
+        # Reconstruct the fileystem hierarchy by building a trie from
```

Comment:
```suggestion
        # Reconstruct the fileystem hierarchy by building a tree from
```

Reply:
"Trie" is another name for "prefix tree", but I will replace it with the latter to avoid confusion.